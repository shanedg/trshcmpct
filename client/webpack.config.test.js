import serializer from 'jest-serializer-path';

import webpackConfig from './webpack.config';

// Remove absolute file paths from snapshots.
expect.addSnapshotSerializer(serializer);

describe('webpackConfig', () => {
  describe('development mode', () => {
    const developmentEnvironment = {};
    const developmentConfig = webpackConfig(developmentEnvironment);
    Object.freeze(developmentConfig);

    it('matches snapshot', () => {
      expect(developmentConfig).toMatchSnapshot();
    });

    it('sets development mode', () => {
      expect(developmentConfig.mode).toEqual('development');
    });

    it('turns off minimize optimization', () => {
      expect(developmentConfig.optimization).toHaveProperty('minimize', false);
    });
  });

  describe('production mode', () => {
    const productionEnvironment = { production: true };
    const productionConfig = webpackConfig(productionEnvironment);
    Object.freeze(productionConfig);

    it('matches snapshot', () => {
      expect(productionConfig).toMatchSnapshot();
    });

    it('sets production mode', () => {
      expect(productionConfig.mode).toEqual('production');
    });

    it('turns on minimize optimization', () => {
      expect(productionConfig.optimization).toHaveProperty('minimize', true);
    });
  });
});
