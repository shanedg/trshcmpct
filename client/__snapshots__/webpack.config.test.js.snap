// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpackConfig development mode matches snapshot 1`] = `
{
  "devServer": {
    "historyApiFallback": true,
    "setupMiddlewares": [Function],
  },
  "devtool": "eval-source-map",
  "entry": "<PROJECT_ROOT>/src/index.ts",
  "mode": "development",
  "module": {
    "rules": [
      {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "options": {
          "configFile": "<PROJECT_ROOT>/babel.config.cjs",
        },
        "test": /\\\\\\.\\(ts\\|tsx\\)\\$/,
      },
      {
        "test": /\\\\\\.css\\$/i,
        "use": [
          "style-loader",
          "css-loader",
        ],
      },
    ],
  },
  "optimization": {
    "minimize": false,
    "minimizer": [
      TerserPlugin {
        "options": {
          "exclude": undefined,
          "extractComments": true,
          "include": undefined,
          "minimizer": {
            "implementation": [Function],
            "options": {
              "warnings": false,
            },
          },
          "parallel": true,
          "test": /\\\\\\.\\[cm\\]\\?js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
        },
      },
    ],
    "splitChunks": {
      "cacheGroups": {
        "defaultVendors": {
          "priority": -10,
          "test": /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]/,
        },
      },
      "chunks": "async",
      "minChunks": 1,
      "minSize": 30000,
    },
  },
  "output": {
    "clean": true,
    "filename": "[name].[chunkhash].js",
    "path": "<PROJECT_ROOT>/dist",
  },
  "plugins": [
    ESLintWebpackPlugin {
      "key": "ESLintWebpackPlugin",
      "options": {
        "cache": true,
        "cacheLocation": "node_modules/.cache/eslint-cache/",
        "emitError": false,
        "emitWarning": true,
        "extensions": [
          ".ts",
          ".tsx",
        ],
        "failOnError": false,
        "lintDirtyModulesOnly": false,
        "reportUnusedDisableDirectives": "warn",
        "resourceQueryExclude": [],
      },
      "run": [Function],
    },
    WebpackManifestPlugin {
      "options": {
        "assetHookStage": Infinity,
        "basePath": "",
        "fileName": "manifest.json",
        "filter": null,
        "generate": undefined,
        "map": null,
        "publicPath": "",
        "removeKeyHash": /\\(\\[a-f0-9\\]\\{16,32\\}\\\\\\.\\?\\)/gi,
        "seed": undefined,
        "serialize": [Function],
        "sort": null,
        "transformExtensions": /\\^\\(gz\\|map\\)\\$/i,
        "useEntryKeys": false,
        "useLegacyEmit": false,
        "writeToFileEmit": false,
      },
    },
    HtmlWebpackPlugin {
      "options": {
        "base": false,
        "cache": true,
        "chunks": "all",
        "chunksSortMode": "auto",
        "compile": true,
        "excludeChunks": [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": "head",
        "meta": {},
        "minify": "auto",
        "publicPath": "auto",
        "scriptLoading": "defer",
        "showErrors": true,
        "template": "<PROJECT_ROOT>/src/index.html",
        "templateContent": false,
        "templateParameters": [Function],
        "title": "trshcmpctr",
        "xhtml": false,
      },
      "userOptions": {
        "template": "<PROJECT_ROOT>/src/index.html",
        "title": "trshcmpctr",
      },
      "version": 5,
    },
  ],
  "resolve": {
    "extensions": [
      ".js",
      ".jsx",
      ".ts",
      ".tsx",
    ],
  },
}
`;

exports[`webpackConfig production mode matches snapshot 1`] = `
{
  "devServer": {
    "historyApiFallback": true,
    "setupMiddlewares": [Function],
  },
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/src/index.ts",
  "mode": "production",
  "module": {
    "rules": [
      {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "options": {
          "configFile": "<PROJECT_ROOT>/babel.config.cjs",
        },
        "test": /\\\\\\.\\(ts\\|tsx\\)\\$/,
      },
      {
        "test": /\\\\\\.css\\$/i,
        "use": [
          "style-loader",
          "css-loader",
        ],
      },
    ],
  },
  "optimization": {
    "minimize": true,
    "minimizer": [
      TerserPlugin {
        "options": {
          "exclude": undefined,
          "extractComments": true,
          "include": undefined,
          "minimizer": {
            "implementation": [Function],
            "options": {
              "warnings": false,
            },
          },
          "parallel": true,
          "test": /\\\\\\.\\[cm\\]\\?js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
        },
      },
    ],
    "splitChunks": {
      "cacheGroups": {
        "defaultVendors": {
          "priority": -10,
          "test": /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]/,
        },
      },
      "chunks": "async",
      "minChunks": 1,
      "minSize": 30000,
    },
  },
  "output": {
    "clean": true,
    "filename": "[name].[chunkhash].js",
    "path": "<PROJECT_ROOT>/dist",
  },
  "plugins": [
    ESLintWebpackPlugin {
      "key": "ESLintWebpackPlugin",
      "options": {
        "cache": true,
        "cacheLocation": "node_modules/.cache/eslint-cache/",
        "emitError": true,
        "emitWarning": false,
        "extensions": [
          ".ts",
          ".tsx",
        ],
        "failOnError": true,
        "lintDirtyModulesOnly": false,
        "reportUnusedDisableDirectives": null,
        "resourceQueryExclude": [],
      },
      "run": [Function],
    },
    WebpackManifestPlugin {
      "options": {
        "assetHookStage": Infinity,
        "basePath": "",
        "fileName": "manifest.json",
        "filter": null,
        "generate": undefined,
        "map": null,
        "publicPath": "",
        "removeKeyHash": /\\(\\[a-f0-9\\]\\{16,32\\}\\\\\\.\\?\\)/gi,
        "seed": undefined,
        "serialize": [Function],
        "sort": null,
        "transformExtensions": /\\^\\(gz\\|map\\)\\$/i,
        "useEntryKeys": false,
        "useLegacyEmit": false,
        "writeToFileEmit": false,
      },
    },
    HtmlWebpackPlugin {
      "options": {
        "base": false,
        "cache": true,
        "chunks": "all",
        "chunksSortMode": "auto",
        "compile": true,
        "excludeChunks": [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": "head",
        "meta": {},
        "minify": "auto",
        "publicPath": "auto",
        "scriptLoading": "defer",
        "showErrors": true,
        "template": "<PROJECT_ROOT>/src/index.html",
        "templateContent": false,
        "templateParameters": [Function],
        "title": "trshcmpctr",
        "xhtml": false,
      },
      "userOptions": {
        "template": "<PROJECT_ROOT>/src/index.html",
        "title": "trshcmpctr",
      },
      "version": 5,
    },
  ],
  "resolve": {
    "extensions": [
      ".js",
      ".jsx",
      ".ts",
      ".tsx",
    ],
  },
}
`;
